app/components/navbar.tsx

```diff
@@ -0,0 +1,94 @@
+import * as React from 'react'
+import type {ReactNode} from 'react'
+import {SunIcon} from './icons/sun-icon'
+import {MoonIcon} from './icons/moon-icon'
+import {MenuIcon} from './icons/menu-icon'
+
+interface NavLinkProps {
+  href: string
+  active?: boolean
+  children?: ReactNode
+}
+
+function NavLink({href, children}: NavLinkProps) {
+  return (
+    <li>
+      <a
+        href={href}
+        className="block px-5 py-2 hover:underline whitespace-nowrap text-lg font-medium"
+      >
+        {children}
+      </a>
+    </li>
+  )
+}
+
+function DarkModeToggle() {
+  const [mode, setMode] = React.useState<'light' | 'dark'>('dark')
+
+  React.useEffect(() => {
+    const classList = document.body.classList
+    classList.remove('light', 'dark')
+    classList.add(mode)
+  }, [mode])
```

Comment:
The effect is just a quick hack, should be replaced someday.

Reply:
I have done a bit of work on this but I'll want to change it I think. Here it is now: https://github.com/kentcdodds/remix-kentcdodds/blob/de8f7ec532d414cfbc2ad1798be16cd28032d734/app/theme-provider.tsx

I think this is something I can handle.